js/forum_italia.js

```diff
@@ -0,0 +1,295 @@
+              commentsCount: topic.posts_count,
+              canPost: (!topic.archived && !topic.closed) ? 'can-post' : ''
+            }, 'forum-italia-comments'))
+
+            forumItalia.commentsMarkup($forumItaliaComment.find(that.topicCommentsSelector), posts);
+            
+            return {
+              $element: $forumItaliaComment,
+              topic: topic,
+            }
+          }).catch(function(error) { /* topic not found or not public */ });
+        });
+
+        Promise.all(forumItaliaComments).then(function(forumItaliaCommentsTopics) {
+          forumItaliaCommentsTopics.map(function(forumItaliaCommentsTopic) {
+            forumItalia.headingMarkup(forumItaliaCommentsTopic.$element.closest('.section'), forumItaliaCommentsTopic.topic.id, forumItaliaCommentsTopic.topic.posts_count);
+          });
+          that.client.isLoggedIn().then(function(isLoggedIn) {
+            if (isLoggedIn) {
+              forumItalia.setLoggedInMarkup();
+            } else {
+              forumItalia.setLoggedOutMarkup();
+            }
+            forumItalia.setClosedTopicsMarkup();
+          });
+        });
+      });
+    },
+
+    headingMarkup: function($section, topicId, postsCount) {
+      $section.find('.chapter-nav__list.chapter-nav__list--visible').first().children('.chapter-nav__item').last().before(tpl({
+        postsCount: postsCount,
+        topicId: topicId
+      }, 'section_navigation__comments'));
+    },
+
+    login: function() {
+      that.client.login().then(function() {
+        forumItalia.setLoggedInMarkup();
+      }).catch(function(error) {
+        console.log('error', error);
+      });
+    },
+
+    logout: function() {
+      that.client.logout().then(function() {
+        forumItalia.setLoggedOutMarkup();
+      });
+    },
+
+    setLoggedInMarkup: function() {
+      // Comment box
+      $(that.commentBoxesSelector)
+        .html(that.forumCommentMarkup)
+        .find(that.userAvatarClass)
+        .attr('src', that.client.getCurrentUserAvatarUrl())
+        .attr('alt', that.client.getCurrentUserDisplayName());
+
+      // Suggestions button click
+      $('.new-comment__suggestions').popover({
+        template: that.suggestionsMarkup,
+        html: true,
+        placement: 'bottom',
+        container: 'body',
+        offset: '175px, 40px',
+        trigger: 'focus'
+      });
+
+      // Logout modal
+      $('body').append(that.logoutModal);
+      $('#logout-modal__submit').click(forumItalia.logout);
+
+      if (that.client.isCurrentUserSilenced()) {
+        return forumItalia.setSilencedMarkup();
+      }
+
+      // Minimum characters
+      $('textarea.box-comment__body').bind('input', function() {
+        $requiredChars = $(this).parents('div.box-comment').find('.required-chars');
+        $requiredMessage = $(this).parents('div.box-comment').find('.box-comment__required');
+        $sendButton = $(this).parents('div.box-comment').find('.box-comment__submit');
```

Comment:
Qui cachare `$(this).parents('div.box-comment')` aiuterebbe un po' le performance.